File: server/iih_pub/iih.ci_pub/src/main/java/iih/ci/ord/i/ortmpl/IOrTmplApplyService.java
package iih.ci.ord.i.ortmpl;

import iih.bd.srv.ortpl.d.OrTmplCaDO;
import iih.bd.srv.ortpl.d.OrTplNItmDO;
import iih.bd.srv.ortpl.dto.OrTmplDTO;
import iih.ci.ord.dto.newordertemplatedto.d.OrderTemplateRstDTO;
import iih.ci.ord.ems.d.CiEnContextDTO;
import xap.mw.core.data.BizException;

/**
 * 医嘱模板加载接口
 * @author Young
 *
 */
public interface IOrTmplApplyService {

	/**
	 * 查询医嘱模板分类
	 * @param id_grp
	 * @param id_org
	 * @param id_dep
	 * @param id_emp
	 * @param sd_ortmpltp
	 * @return
	 * @throws BizException
	 */
	public OrTmplCaDO[] getOrTmplCaDOs(String id_grp, String id_org, String id_dep, String id_emp, String sd_ortmpltp) throws BizException;
	/**
	 * 查询医嘱模板分类关系
	 * @param id_ortmplca
	 * @param sd_ortmpltp
	 * @return
	 * @throws BizException
	 */
	public OrTmplDTO[] getOrTmplDTOs(String id_ortmplca,String sd_ortmpltp) throws BizException;
	/**
	 * 查询医嘱模板明细
	 * @param id_ortmpl
	 * @return
	 * @throws BizException
	 */
	public OrTplNItmDO[] getOrTplNItmDOs(String id_ortmpl) throws BizException;
	/**
	 * 查询医嘱模板明细
	 * @param id_ortmpls
	 * @param ctx
	 * @return
	 * @throws BizException
	 */
	public OrderTemplateRstDTO getOrTemplateCache(String[] id_ortmpls, CiEnContextDTO ctx) throws BizException;
}

File: client/iih_stub/iih.ci_stub/iih.ci.ord/i/ortmpl/IOrTmplApplyService.cs

using System;
using iih.bd.srv.ortpl.d;
using iih.bd.srv.ortpl.dto;
using iih.ci.iih.ci.ord.dto.newordertemplatedto.d;
using iih.ci.ord.ems.d;

namespace iih.ci.ord.i.ortmpl
{
    /// <summary>
    /// <para>描    述 :  医嘱模板加载服务接口			</para>
    /// <para>说    明 :  医嘱模板加载服务接口   			</para>
    /// <para>项目名称 :  iih.ci.ord.i.ortmpl    </para>    
    /// <para>类 名 称 :  IOrTmplApplyService					</para> 
    /// <para>版 本 号 :  v1.0.0.0           			</para> 
    /// <para>作    者 :  Young         				</para> 
    /// <para>修 改 人 :  Young         				</para> 
    /// <para>创建时间 :  2017/10/19 15:16:39             </para>
    /// <para>更新时间 :  2017/10/19 15:16:39             </para> 
    /// <para>Copyright @ 北大医信（IIH项目组） 2017. All rights reserved.</para> 
    /// </summary>
    public interface IOrTmplApplyService
    {
        /// <summary>
        /// 查询医嘱模板分类
        /// </summary>
        /// <param name="id_grp"></param>
        /// <param name="id_org"></param>
        /// <param name="id_dep"></param>
        /// <param name="id_emp"></param>
        /// <param name="sd_ortmpltp"></param>
        /// <returns></returns>
        OrTmplCaDO[] getOrTmplCaDOs(String id_grp, String id_org, String id_dep, String id_emp, String sd_ortmpltp);
	
        /// <summary>
        /// 查询医嘱模板
        /// </summary>
        /// <param name="id_ortmplca"></param>
        /// <param name="sd_ortmpltp"></param>
        /// <returns></returns>
        OrTmplDTO[] getOrTmplDTOs(String id_ortmplca,String sd_ortmpltp);

        /// <summary>
        /// 查询模板明细
        /// </summary>
        /// <param name="id_ortmpl"></param>
        /// <returns></returns>
        OrTplNItmDO[] getOrTplNItmDOs(String id_ortmpl);

        /// <summary>
        /// 查询模板明细
        /// </summary>
        /// <param name="id_ortmpl"></param>
        /// <param name="ctx"></param>
        /// <returns></returns>
        OrderTemplateRstDTO getOrTemplateCache(String[] id_ortmpl, CiEnContextDTO ctx);
    }
}
